llms/mlx_lm/utils.py

```diff
@@ -174,6 +220,12 @@ def generate(
                 print(s[skip:], end="", flush=True)
                 skip = len(s)
 
+    repetition_context.append(
+        token[0].item()
+    )  # Update repetition context after each token
+    repetition_context = repetition_context[
+        -repetition_context_size:
+    ]  # Maintain the specified context size
```

Comment:
Issue: The update repetition context should be inside the loop.
```suggestion
      repetition_context.append(
          token[0].item()
      )  # Update repetition context after each token
      repetition_context = repetition_context[
          -repetition_context_size:
      ]  # Maintain the specified context size
```